pkg/cloud/vsphere/provisioner/common/templates.go

```diff
@@ -406,8 +406,27 @@ KUBELET=$(getversion kubelet ${KUBELET_VERSION}-)
 KUBEADM=$(getversion kubeadm ${KUBELET_VERSION}-)
 KUBECTL=$(getversion kubectl ${KUBELET_VERSION}-)
 apt-get install -y docker.io=${DOCKER_VER}
-# Explicit cni version is a temporary workaround till the right version can be automatically detected correctly
-apt-get install -y kubelet=${KUBELET} kubeadm=${KUBEADM} kubectl=${KUBECTL}
```

Comment:
While debugging this issue with @sflxn  seems like the upstream repo has for now the 0.7.5 version of the kubernetes-cni available for apt-get install.

Thus we would fall back to explicitly choosing the kubernetes-cni version based on the kubernetes version we are trying to install.

Reply:
ideally CNI 0.7.5 should be installed in any case for anything > 1.11, because it includes a security fix.

also i would like to point out that the packages are a "best effort" and in a flux, thus if you are pinning your base distro to Ubuntu already, you might as well just download the k8s binaries, cni binaries and systemd unit files and not rely on the packages until they are in a better shape.

Reply:
@neolit123 I noticed when I install kubernetes-cni 0.7.5 manually, it also installs kubelet and kubeadm 1.14.

Reply:
> I noticed when I install kubernetes-cni 0.7.5 manually, it also installs kubelet and kubeadm 1.14.

hm, i cannot confirm this on Debian Stretch:

```
$ apt-get install kubernetes-cni=0.7.5-00
Reading package lists... Done
Building dependency tree       
Reading state information... Done
The following NEW packages will be installed:
  kubernetes-cni
0 upgraded, 1 newly installed, 0 to remove and 0 not upgraded.
Need to get 6473 kB of archives.
After this operation, 50.2 MB of additional disk space will be used.
Get:1 https://packages.cloud.google.com/apt kubernetes-xenial/main amd64 kubernetes-cni amd64 0.7.5-00 [6473 kB]
Fetched 6473 kB in 3s (1918 kB/s)         
debconf: delaying package configuration, since apt-utils is not installed
Selecting previously unselected package kubernetes-cni.
(Reading database ... 7341 files and directories currently installed.)
Preparing to unpack .../kubernetes-cni_0.7.5-00_amd64.deb ...
Unpacking kubernetes-cni (0.7.5-00) ...
Setting up kubernetes-cni (0.7.5-00) ...
```

again, i do encourage the vsphere provider to do this:

> ... just download the k8s binaries, cni binaries and systemd unit files and not rely on the packages until they are in a better shape.

the process i'm talking about is similar to what we are doing here for container linux:
https://kubernetes.io/docs/setup/independent/install-kubeadm/#installing-kubeadm-kubelet-and-kubectl

except that the versions need a bump and also these complimentary non-go dependencies are nice-to-have:
https://github.com/kubernetes/release/blob/master/debian/xenial/kubelet/debian/control#L13

Reply:
I agree with the approach to use the binaries directly.
However, the only question is do we want to do that right way in`v1alpha1`  or maybe do that after the milestone.
My opinion is handle this change post `v1apha1` but I will let @sflxn also chime in here.

Reply:
I think this is a temporary fix to allow us to get on with other issues.  I think we will need to revisit how we install packages post v1alpha1.  This may get changed within CAPI so I don't want to make such a radical changes that might get ripped out in a few months.